### DISCUSSION TITLE
Tdarr and intel iGPU access?

### DISCUSSION THREAD

**Discussion by mervincm** [tteck/Proxmox#1765] [Help Needed] [answered]
I installed tdarr w `bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/ct/tdarr.sh)"`
I picked the defaults other than specifying the appropriate bridge and it starts / runs as expected. I had to enable NFS and install NFS-common to access my media share but that is now done. My question is will I have access to my host's intel N100 iGPU by default (it appears not) or is this something that is after the fact config. any advice is appreciated.

**Comment by mervincm** [comment]
I should note that the guides I found point to entries in the lxc conf file, and it appears your script pre-populates all of the suggested entries. it is just that tdarr doesn't seem to see/use it.

**Comment by mervincm** [comment]
I suspect it might just be tdarr, since when I use lxc console to install the intel GPU tools with apt install intel-gpu-tools, and then run intel_gpu_top, it works just fine, displaying GPU activity being done on my plex lxc.

**Comment by Dulanic** [comment]
So did you confirm this is using quicksync? I was just going to move my tdarr over to LXC, and wsnt sure if it was working.

**Reply by mervincm** [reply]
I added handbrake but was not able to get tdarr to use the available GPU via LXC. I know plex used it (confirmed) but no luck with LXC. I moved the host to Debian and local docker and it works as suspected.

**Reply by Dulanic** [reply]
Same, I have it still in docker and no issues, but LXC just sits there if I tell it to only use GPU.

**Reply by migueljal** [reply]
I have discovered why the LXC installation does not work with VAAPI. Use the fmmpeg version:

ffmpeg version 5.0.1-static https://johnvansickle.com/ffmpeg/ Copyright (c) 2000-2022 the FFmpeg developers
   built with gcc 8 (Debian 8.3.0-6)
   configuration: --enable-gpl --enable-version3 --enable-static --disable-debug --disable-ffplay --disable-indev=sndio --disable-outdev=sndio --cc=gcc --enable- fontconfig --enable-frei0r --enable-gnutls --enable-gmp --enable-libgme --enable-gray --enable-libaom --enable-libfribidi --enable-libass --enable-libvmaf --enable- libfreetype --enable-libmp3lame --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopenjpeg --enable-librubberband --enable-libsoxr --enable-libspeex --enable-libsrt --enable-libvorbis --enable-libopus --enable-libtheora --enable-libvidstab --enable-libvo-amrwbenc --enable-libvpx --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxml2 --enable -libdav1d --enable-libxvid --enable-libzvbi --enable-libzimg

When the docker version uses:

ffmpeg version 6.0-Jellyfin Copyright (c) 2000-2023 the FFmpeg developers
   built with gcc 9 (Ubuntu 9.4.0-1ubuntu1~20.04.2)
   configuration: --prefix=/usr/lib/jellyfin-ffmpeg --target-os=linux --extra-version=Jellyfin --disable-doc --disable-ffplay --disable-ptx-compression --disable-static --disable-libxcb --disable-sdl2 --disable-xlib --enable-lto --enable-gpl --enable-version3 --enable-shared --enable-gmp --enable-gnutls --enable-chromaprint --enable-libdrm --enable-libass --enable-libfreetype --enable-libfribidi --enable-libfontconfig --enable-libbluray --enable-libmp3lame --enable-libopus --enable-libtheora --enable-libvorbis --enable-libopenmpt --enable-libdav1d --enable-libwebp --enable-libvpx --enable-libx264 --enable-libx265 --enable-libzvbi --enable-libzimg --enable-libfdk-aac --arch =amd64 --enable-libsvtav1 --enable-libshaderc --enable-libplacebo --enable-vulkan --enable-opencl --enable-vaapi --enable-amf --enable-libvpl --enable-ffnvcodec --enable -cuda --enable-cuda-llvm --enable-cuvid --enable-nvdec --enable-nvenc

**Reply by tteck** [reply, marked as answer]
Have you tested it with the new ffmpeg version?

```
wget -q https://github.com/jellyfin/jellyfin-ffmpeg/releases/download/v6.0.1-1/jellyfin-ffmpeg6_6.0.1-1-bookworm_amd64.deb
dpkg -i jellyfin-ffmpeg6_6.0.1-1-bookworm_amd64.deb
```

**Reply by migueljal** [reply]
Now it works! Thank you!

**Reply by networkoctopus** [reply]
I too continue to have issues with this, also have n100 mini-PC. 

I am new to tdarr and ffmpeg so maybe I am missing a step here. 
I followed the above additional steps to add alternate ffmpeg, but my tdarr continues to use the previous version, 5.0.1. 

Is there a config change I have to make in tdarr to use the new ffmpeg?
Thanks!

**Reply by networkoctopus** [reply]
Okay I figured it out.  I couldn't find any good instructions w google.. Maybe below is assumed knowledge.
Incase anyone needs help figuring out how to point tdarr to the above installed jellyfin ffmpeg - you simply need to edit your node/server config file. nano Tdarr_Node_Config.json, Tdarr_Server_Config.json
Mine were in /opt/tdarr/configs.

My Jellyfin ffmpeg installed to /usr/lib/jellyfin-ffmpeg/ffmpeg, so I added the path to my config files as below.

{
  "nodeName": "tan-topi",
  "serverIP": "0.0.0.0",
  "serverPort": "8266",
  "handbrakePath": "",
  "ffmpegPath": "/usr/lib/jellyfin-ffmpeg/ffmpeg",
  "mkvpropeditPath": "",
  "pathTranslators": [
    {
      "server": "",
      "node": ""
    }
  ],
  "logLevel": "INFO",
  "priority": -1,
  "cronPluginUpdate": ""
}

Once tdarr starts with new config, open tdarr webpage and head over to the help tab - click run and if all is well it should print out:
ffmpeg version 6.0.1-Jellyfin Copyright (c) 2000-2023 the FFmpeg developers
...
...

**Reply by nicoska84** [reply]
So, with this solution it seems to work:
![image](https://github.com/tteck/Proxmox/assets/73702791/faddd3f0-8c28-4a9a-b14c-04bf2bcc0f8f)

But I can see also that my CPU usage is increasing anyway:
![image](https://github.com/tteck/Proxmox/assets/73702791/aab9e6df-5768-408c-933f-27d3b2d96ab3)

**Reply by nicoska84** [reply]
But also, not so bad :)
![image](https://github.com/tteck/Proxmox/assets/73702791/06902b75-8d6a-4214-a838-7131a8f45de6)

**Reply by networkoctopus** [reply]
Yeh same - though I think the cpu usage you're seeing is typical and to be expected, from what I've read. 
Also, audio transcoding is meant to be CPU bound, so there will always be a hit for that.

**Comment by chicungunya** [comment]
@tteck 
Can you please tell me what I miss? I have nothing in my intel_gpu_top. (I use QuickSync)

![image](https://github.com/tteck/Proxmox/assets/25926287/635cd6df-94a2-4e6c-a65c-08c6fbcd0f81)

I used your script to install Tdarr.

When I transcode :

![image](https://github.com/tteck/Proxmox/assets/25926287/4d5643b5-e062-4eb7-82d2-16fb71968412)

So I'm not using CPU right? I mean if I have nothing in intel-gpu-top I don't use it for me.

But when I check the logs it seems to work that's weird?

![image](https://github.com/tteck/Proxmox/assets/25926287/11913cb8-a386-4301-9591-4db8ac5dc1d6)

And for a 2 gb files, it seems really long no? 

![image](https://github.com/tteck/Proxmox/assets/25926287/382a91c8-71fe-463b-abe5-fdf59769336d)


Thanks!

**Reply by networkoctopus** [reply]
you're using a CPU plugin in your workflow (FFMPEG_CPU). Switch to either plugin 'Boosh-Transcode using QSV GPU & FFMPEG' or 'FFMPEG VAAPI HEVC Transcode'.  Both available in the community tab.

<img width="569" alt="Screenshot 2024-03-01 at 09 41 15" src="https://github.com/tteck/Proxmox/assets/77534662/01df76d5-e4db-4a12-b497-9923d734d263">


Also, set CPU transcode workers to 0 on your tdarr node.  This works for me.

<img width="512" alt="Screenshot 2024-03-01 at 09 38 31" src="https://github.com/tteck/Proxmox/assets/77534662/a282996a-db86-40c5-a5df-c2c71b55b2f5">

**Reply by chicungunya** [reply]
Thanks but I don't have a GPU, only CPU with Intel QuickSync. So it won't work right?

**Reply by networkoctopus** [reply]
Quicksync runs on the integrated GPU that's onboard your CPU..

**Reply by chicungunya** [reply]
@dwyeraidan thanks it works!!!

**Comment by mervincm** [comment]
> Have you tested it with the new ffmpeg version?
> 
> ```
> wget -q https://github.com/jellyfin/jellyfin-ffmpeg/releases/download/v6.0.1-1/jellyfin-ffmpeg6_6.0.1-1-bookworm_amd64.deb
> dpkg -i jellyfin-ffmpeg6_6.0.1-1-bookworm_amd64.deb
> ```

In the months since they have rev'd the FFMPEG version a bit, so I used this today.

step 1
```
wget -q https://github.com/jellyfin/jellyfin-ffmpeg/releases/download/v6.0.1-5/jellyfin-ffmpeg6_6.0.1-5-bookworm_amd64.deb
dpkg -i jellyfin-ffmpeg6_6.0.1-5-bookworm_amd64.deb
```
step 2
I then installed it to /usr/lib/jellyfin-ffmpeg/ffmpeg  with
`apt-get install -f ./jellyfin-ffmpeg6_6.0.1-5-bookworm_amd64.deb`

step 3
tell tdarr to use it with (stolen from above with love)
edit the path line in your ( in /opt/tdarr/configs) node/server config files Tdarr_Node_Config.json Tdarr_Server_Config.json
`"ffmpegPath": "/usr/lib/jellyfin-ffmpeg/ffmpeg",`

**Comment by fcps3** [comment]
@dwyeraidan 
Hi, thanks for explaining so clearly, I now use hardware acceleration with intel N100, but something is wrong I think, because I get only 50 fps
![Screenshot 2024-04-12 alle 09 46 20](https://github.com/tteck/Proxmox/assets/90822872/4c7431f7-c006-4584-ad81-a8f2546c03f4)
![Screenshot 2024-04-12 alle 09 45 57](https://github.com/tteck/Proxmox/assets/90822872/b915facb-ea66-4350-940f-822f3785d3a2)
![Screenshot 2024-04-12 alle 09 46 37](https://github.com/tteck/Proxmox/assets/90822872/1fe0d534-0409-4d5a-924e-a1d902ccf586)

**Reply by networkoctopus** [reply]
Have you tried 'Boosh-Transcode using QSV GPU & FFMPEG' plugin instead? I use that one.. you can set the speed preset with its settings - I tested all of them and found no loss in quality using the 'fastest' preset. These days quick sync is so optimised for h.265 there's no need for slow profile anymore. For 1080p it's usually >200fps I find.

**Reply by fcps3** [reply]
Thanks a lot, it works, just another question, is it normal that I see "OS CPU" at 90/100% but on proxmox it says about 2%? Can or should this be improved?

**Reply by mervincm** [reply]
In my experience once you see you activity your set, everything after that is the tool you use and the encoding settings. 50 frames is not an indicator of a problem as long as you see GPU activity. High CPU is also not an indicator of a problem as long as you see gpu activity. If you want it to go faster, or use less cpu then "do less" by picking encoding settings that are less work. Ps I recently read that there is a bug with qsv and quality choice not being respected because of FFMPEG bugs so test after change (on same piece of media!), make sure it's faster.

**Comment by jproz** [comment]
@tteck - re this thread and apologies if you've answered this elsewhere. 
Have you considered making the jellyfin-ffmpeg and mkvpropedit (mkvtoolsnix) a standard part of Tdarr build (including updating the json files accordingly for Tdarr)?
It seems to be inline with using known good packages and, at least those of us using qsv, generally have to do this anyway since vaapi isn't in the packaged ffmpeg with Tdarr. 

(This came up for me again as I rebuilt a bunch of stuff from scratch lately and had totally forgot about that manual step until did some digging.. again)
